### ISSUE TITLE
Installing WLS2 - wsl : The term 'wsl' is not recognized

### ISSUE THREAD

**Issue by Gedili** [microsoft/WSL#6369]
# Environment

Microsoft Windows [Version 10.0.18363.1198]

# Steps to reproduce

I tried to install WSL2 as shown in https://docs.microsoft.com/en-us/windows/wsl/install-win10, following the manual installation instructions.

All the steps went fine, no error, until Step 5:
`wsl --set-default-version 2`
which returned:
```
wsl : The term 'wsl' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the
spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ wsl --set-default-version 2
+ ~~~
    + CategoryInfo          : ObjectNotFound: (wsl:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
```

#  Expected behavior

_The command being recognized and changing the default version to 2_

# Actual behavior

I had that same error (command not recognized) both using the PowerShell (on standard and admin mode), and with cmd.exe
Weirdly, if I open windows explorer in "C:\Windows\System32" I do see a file named wsl.exe (not even having hiden flag). When moving to that same directory from powershell (user or admin) and cmd, this file doesn't even appear with the DIR command:
With CMD:
```
C:\Windows\System32>dir | findstr wsl
19/03/2019  01:45           166 400 windowslivelogin.dll
```
With Powershell:
```
PS C:\WINDOWS\system32> dir | findstr wsl
-a----       19/03/2019     01:45         166400 windowslivelogin.dll
```
However, installing Windows Terminal app (https://docs.microsoft.com/en-us/windows/terminal/get-started), and launching the command from there (which uses the Powershell as well), then the files do show up.
```
PS C:\Windows\System32> dir | findstr wsl
-a----       19/03/2019     01:44         226816 windowslivelogin.dll
-a----       10/12/2020     21:13         201728 wsl.exe
-a----       10/12/2020     21:13          37888 wslapi.dll
-a----       10/12/2020     21:13          76288 wslconfig.exe
```
They also show up in Windows Explorer, as mentioned previously, and in Cygwin as well.

I was able to finish the install from the windows terminal, setting the default version to 2, and installing Ubuntu from the Windows Store, which works fine, and can be ran from the Ubuntu app, or from the Windows Terminal. 

Yet I can't explain why the wsl program isn't found from the standard cmd and powershell, since it's not a problem of executables paths, it doesn't even "see" the file when being in System32 folder.

![mintty_2020-12-22_13-25-43](https://user-images.githubusercontent.com/45516124/102914522-2cda0680-445f-11eb-8fce-285c2fd91f9d.png)

**Comment by DHowett** [comment]
Can you run the following from the both the working and non-working powershell/cmd instances?

* cmd: `echo %PROCESSOR_ARCHITECTURE%`
* powershell: `$Env:PROCESSOR_ARCHITECTURE`

How are you launching the instances that do not work? What sort of machine are you using?

**Comment by Gedili** [comment]
> Can you run the following from the both the working and non-working powershell/cmd instances?
> 
> * cmd: `echo %PROCESSOR_ARCHITECTURE%`
> * powershell: `$Env:PROCESSOR_ARCHITECTURE`
> 
> How are you launching the instances that do not work? What sort of machine are you using?

Oh yeah, seems you're onto something.
I run an Acer Laptop, with an Intel Core i5 (64bits), and the OS, Win 10, is also 64bits.

So from the CMD (where WLS doesn't not work) I get:
```
C:\Windows\System32>echo %PROCESSOR_ARCHITECTURE%
x86
C:\Windows\System32>wmic OS get OSArchitecture
OSArchitecture
64-bit
```

From the standard PowerShell (where WLS doesn't work neither):
```
PS C:\Users\adrien> $Env:PROCESSOR_ARCHITECTURE
x86
PS C:\Users\adrien> reg query "HKLM\SYSTEM\CurrentControlSet\Control\Session Manager\Environment" /v PROCESSOR_ARCHITECTURE
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager\Environment
    PROCESSOR_ARCHITECTURE    REG_SZ    AMD64
```
On both it does show x86, whereas on the PowerShell from the Windows Terminal app (the one that worked) it shows x64:
PS C:\Users\adrien> $Env:PROCESSOR_ARCHITECTURE
AMD64

Here are the full info about the CPU (from lscpu)
```
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   39 bits physical, 48 bits virtual
CPU(s):                          8
On-line CPU(s) list:             0-7
Thread(s) per core:              2
Core(s) per socket:              4
Socket(s):                       1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           142
Model name:                      Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz
Stepping:                        12
CPU MHz:                         2112.006
```

What I don't get is that the x86 shell might not be able to run the WSL program if it's only for 64bit, but shouldn't they at least be able to find the file in System32 using the command DIR ??

**Comment by DHowett** [comment]
So! When you’re running a 32-bit process on 64-bit Windows (or on ARM64, any non-native combination), the hosted non-native process _actually sees a different System32 folder(!)_. The one a 32-bit process sees on x64 Windows is stored on disk at \Windows\SysWOW64. You can access the “native” system32 from a hosted process by looking in the virtual folder \Windows\sysnative. It won’t actually be present on disk, mind you, but the filesystem path resolver will find it. 

The WSL executable is (for better or for worse) only installed to the native system directory. Probably because on paper every binary in the hosted one should match the hosted architecture, and it wouldn’t make sense to ship a 32-bit wsl.exe :smile:

**Comment by Gedili** [comment]
Oh I see, yeah didn't know about that specificity, I assumed you couldn't execute the program, but that you'd at least be able to list the files present in the directory. Thanks, that was puzzling me a lot despite having found a workaround!

And I actually realized why it gave the error in the first place. In "Start Menu\Programs\Windows PowerShel\" there are two shortcuts for the PowerShell, the standard one, and the x86, it was the standard one (supposed to be x64) that I launched, but I actually was launching it from Launchy (https://www.launchy.net/), a small utility to run programs (similar to Alt+F2 on linux). And that software being 32bit, it'd launched the programs in 32bits as well. When I tried launching CMD and Powershell directly from the start menu it worked (I didn't have the issue with the Windows Terminal, because that one I had to launch from the Start Menu)

Thanks again!
